### ISSUE TITLE
First time build issue

### ISSUE THREAD

**Issue by dswilson4** [MetaMask/metamask-mobile#6158]
### **Description**

Was building locally for the first time today but am running into issues that I'm looking to see if others have also encountered recently.

### **Technical Details**

Environment

- Mac OS: Ventura 13.3.1
- Xcode version 14.3 & Command line tools 14.3
- IOS Device: Iphone 12 pro, IOS 16.4 (latest)

1. The first error with the build I’m seeing is :

```
ld: file not found: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/arc/libarclite_iphonesimulator.a
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

It seems the newest version of Ventura has this issue where the arc folder is no longer present where expected, here’s a relevant thread with the resolution I’ve followed (copy previous libarclite_iphonesimulator.a file into new `arc` folder): [https://developer.apple.com/forums/thread/727527](https://developer.apple.com/forums/thread/727527)

2. After resolving the first issue, I’m seeing this build error error with `yoga` in React Native 0.66.5:

```
node_modules/react-native/ReactCommon/yoga/yoga/Yoga.cpp:2232:9: error: use of bitwise '|' with boolean operands [-Werror,-Wbitwise-instead-of-logical]
node->getLayout().hadOverflow() |
```

This issue was raised in early February tracked here in FB’s RN repo: [https://github.com/facebook/yoga/issues/1224](https://github.com/facebook/yoga/issues/1224)

The issue was fixed in April 2022, but versions of RN > 0.66.   Should developers instead use an older version of XCode assuming the intended Metamask RN version is 0.66.5?

### **Alternative**
I also tried uninstalling Xcode 14 and installing Xcode 13 instead, but it is incompatible with Ventura OS.  Does anybody else see any of the above issues when trying to build on a clean machine and test on an IOS simulator?

**Comment by YDisa** [comment]
I am on the same issue.

After the xcode update,
My client's old app doesn't build
(Builded before update)

---
xcode : Version 14.3 (14E222b)
react-native : 0.66.2

**Comment by dswilson4** [comment]
@YDisa I haven't made any progress so far

**Comment by wantaekchoi** [comment]
same problem with me

**Comment by roti-c137** [comment]
Temporary fix is to click on the error to auto navigate to the file and line of code, and make the fix as suggested

**Comment by wantaekchoi** [comment]
> Temporary fix is to click on the error to auto navigate to the file and line of code, and make the fix as suggested

I tried fixing the issue by changing  `|` -> `||` in Yoga.cpp:2232:9 (ref: https://github.com/facebook/react-native/commit/52d8a797e7a6be3fa472f323ceca4814a28ef596 (https://github.com/facebook/yoga/commit/f174de70afdde2492e8677bd0e716eb41bf64469).

However, I encountered another error:
```bash
.../Pods.build/Debug-iphonesimulator/YogaKit.build/Objects-normal/arm64/YogaKit.swiftinterface:6:19: error: underlying Objective-C module 'YogaKit' not found
@_exported import YogaKit
```

Do you have any suggestions for me?

**Comment by det-joe** [comment]
> 

Mac OS: Ventura 13.2.1
Xcode version 14.3
<img width="1109" alt="スクリーンショット 2023-05-29 17 58 02" src="https://github.com/MetaMask/metamask-mobile/assets/82364464/4b91c103-ac82-4907-a6fd-89915b62d7f1">

Same issue here, fixed the  | -> || but got YogaKit not found issue like @wantaekchoi

**Comment by ZainaliSyed** [comment]
xcode : Version 14.3 (14E222b)
react-native : 0.66.2

I am also facing error either running via **xcode**  or using **yarn start:ios**
 
![Screenshot 2023-06-06 at 5 12 59 PM](https://github.com/MetaMask/metamask-mobile/assets/24393023/cef18fb0-02f7-4b10-b06c-751c3640cdbc)

![Screenshot 2023-06-06 at 5 14 00 PM](https://github.com/MetaMask/metamask-mobile/assets/24393023/2754f56f-5966-41c7-bb42-5df01fae9086)

**Comment by vsradical** [comment]
Bumping this issue for support. I've tried the suggestions in this post and #6418
Nothing has yet to work.

Does anybody have any work-arounds to temporarily get this working on an iphone with the latest iOS version?

**Comment by sethkfman** [comment]
We are looking into the issue. The current recommendation is to revert back to Xcode 14.2. 

@leotm I know you were looking into this issue. Did you have any progress?

**Comment by leotm** [comment]
thanks for looking into this ^ ran into similar (looking slightly ahead in the future)
- https://github.com/MetaMask/metamask-mobile/pull/6220#pullrequestreview-1479503045
  - _nb: includes https://github.com/facebook/react-native/commit/52d8a797e7a6be3fa472f323ceca4814a28ef596_

tl;dr so far
- unrelated to Flipper (despite pod deps on YogaKit and Yoga)
- unrelated to project/YogaKit Swift 4/5 module interfaces (warnings)
- unrelated to Xcode Project Document Format 12/13/14/15-compat
- YogaKit/yoga not found persists on Xcode 15 beta 1
- stick to Xcode 14.2 till we find a resolution to https://developer.apple.com/forums/thread/728161

next steps
- test Xcode 15 beta 2 (released yday 21Jun)
- test RN 0.72 (requires branching off 0.71.6 PR above and then [upgrading](https://github.com/reactwg/react-native-releases/discussions/74#discussioncomment-6245308))
- continue monitoring/deep-dive into https://developer.apple.com/forums/thread/728161

**Comment by bradherman** [comment]
Getting same issue in Xcode 15 beta 5

**Comment by leotm** [comment]
notes from fresh investigation on `main` 9ccfbc3

download [Xcode 15b8](https://developer.apple.com/services-account/download?path=/Developer_Tools/Xcode_15_beta_8/Xcode_15_beta_8.xip) (3.35 GB), extract, `xcode-select --switch /Applications/Xcode-beta.app`, `xcodebuild -license accept`, `xcodebuild -runFirstLaunch`

_building on: iPhone 14 Pro (iOS 16.1) and iOS SE (3rd generation) (iOS 15.4)_

build error: `iOS 17.0 is not installed`
run Xcode and GET iOS 17 (7.5 GB)

<img width="1056" alt="Screenshot 2023-08-31 at 16 52 25" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/fdcc5ff5-0e82-4933-9425-d530980b48d1">

note similar issue from the error
- https://github.com/CocoaPods/CocoaPods/issues/12033

bump `ios/branch-ios-sdk`: `IPHONEOS_DEPLOYMENT_TARGET = 8.0;` to `12.0` to fix

<img width="1061" alt="Screenshot 2023-08-31 at 16 52 02" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/a400d6f0-3c2e-4990-9942-392d26735569">

bump `ios/mixpanel-iphone`: `IPHONEOS_DEPLOYMENT_TARGET = 8.0;` to `12.0` to fix

<img width="1057" alt="Screenshot 2023-08-31 at 16 55 11" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/a78609be-1eeb-47ee-8f92-0ca1ed8cc2cc">

noting similar issue from the error
- https://github.com/facebook/react-native/issues/37748

update Pods config to fix

<img width="1327" alt="Screenshot 2023-08-31 at 17 01 32" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/f2fd2c2a-0b7a-41da-ad1d-3804b4271121">

now we get our original error
https://developer.apple.com/forums/thread/728161

<img width="1322" alt="Screenshot 2023-08-31 at 17 03 26" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/afe51b82-5530-4526-9136-254c8d5b166c">

i think at this point the next safest best would be to upgrade to RN 0.71.13 (cc @jpcloureiro) and re-test

then if to no avail repro on RN 0.71.13 minimising deps

also curious @cipolleschi if there's anything obvious we're missing

**Comment by claudioitalian12** [comment]
not fixed bumping @sentry deps since no Pod lockfile RNSentry/Sentry/SentryPrivate changes

TODO bump node_modules/@sentry/react-native >   s.ios.deployment_target = "11.0" to 12.0
TODO revise project IPHONEOS_DEPLOYMENT_TARGET = 11.0

I have try to do this last step but not working, where i need change IPHONEOS_DEPLOYMENT_TARGET = 11.0?

**Comment by leotm** [comment]
in the draft pr
- https://github.com/MetaMask/metamask-mobile/pull/7114

spotted conflicts to resolve related to Sentry, so reverted my changes (https://github.com/MetaMask/metamask-mobile/commit/a7ec533c98c03bff3b7f700b19bf646d671e5178 interestingly no pod changes at the time on Xcode 15 beta 8) and pulled in `main` (accepting incoming changes including https://github.com/MetaMask/metamask-mobile/pull/7277) to update the pr branch

since then noticing on Xcode 15 stable

<img width="1400" alt="Screenshot 2023-10-02 at 3 57 48 pm" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/def93fea-037b-4828-8036-886e22e445eb">

is no longer fixed by https://github.com/facebook/react-native/issues/37748#issuecomment-1580334650, but updating `unary_function` to `__unary_function` works

then https://github.com/MetaMask/metamask-mobile/pull/7114/commits/0d199d7e8efb1c2e8f6988fe9092c8a5fae795bf and https://github.com/MetaMask/metamask-mobile/pull/7114/commits/acb6d40c94db4e3fe5d56f0ee8747f842bed8f9f needed reverting to fix CI (postinstall `git submodule update --init` final step 5)

after that ran into another error https://github.com/MetaMask/metamask-mobile/pull/7114#discussion_r1342661166 fixed in https://github.com/MetaMask/metamask-mobile/commit/a09d517135af02474bbef3e461a7e50d32f99830

but then Branch and MixPanel `IPHONEOS_DEPLOYMENT_TARGET` bumps are required again (versioned via gitsubmodule commits), but appear problematic with our CI postinstall step, so there may be a better way to upgrade these

now encountering the same _SentryPrivate' not found_ error, but this time with the warning has been resolved

<img width="1628" alt="Screenshot 2023-10-02 at 4 04 42 pm" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/6b88c4a9-be1b-40f2-a06a-9fd317c3eeb9">

**Comment by leotm** [comment]
thanks for checking again  @claudioitalian12 ^

raised the issue earlier
- https://github.com/getsentry/sentry-react-native/issues/3322

and got to a working solution thanks to @krystofwoldrich

detailed here https://github.com/getsentry/sentry-react-native/issues/3322#issuecomment-1750805368

with 3 final TODO's noted we have remaining to tackle
